server/core/region_tree_test.go

```diff
@@ -302,3 +347,18 @@ func BenchmarkRegionTreeUpdate(b *testing.B) {
 		tree.update(item)
 	}
 }
+
+const MaxKey = 10000000
+
+func BenchmarkRegionTreeUpdateUnordered(b *testing.B) {
+	tree := newRegionTree()
+	var keys []int
+	for i := 0; i < MaxKey; i++ {
+		keys = append(keys, rand.Intn(MaxKey))
+	}
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		item := &RegionInfo{meta: &metapb.Region{StartKey: []byte(fmt.Sprintf("%20d", keys[i])), EndKey: []byte(fmt.Sprintf("%20d", keys[i+1]))}}
```

Comment:
It is possible that `keys[i+1] < keys[i]`, then this will be an invalid region.
What's more, maybe move this line outside the loop will make this benchmark more accurate.